cost_calculator/instance_selector.py

```diff
@@ -176,12 +194,69 @@ def get_cheapest_instance(self, cpu_request, memory_request, gpu_spec):
         lowest_price = 100000000.0
         for inst in matches:
             price, is_t_unlimited = self.price_for_cpu_spec(cpu_request, inst)
-            if price > 0.0 and price < lowest_price:
+            if 0.0 < price < lowest_price:
                 lowest_price = price
                 cheapest_instance = inst['instanceType']
                 if is_t_unlimited:
                     cheapest_instance += ' (unlimited)'
-        return cheapest_instance, lowest_price
+        lowest_spot_price = lowest_price
+        lowest_spot_price = min(lowest_spot_price, self.get_spot_price(cheapest_instance))
+        return cheapest_instance, lowest_price, lowest_spot_price
+
+
+class PriceGetter:
+    provider_keys_map = {
+        "azure": "azure",
+        "aws": "amazon",
+        "gce": "google"
+    }
+
+    def __init__(self, provider, redis_client):
+        self.provider = provider
+        self.redis_client = redis_client
+        self.key_pattern = "/banzaicloud.com/cloudinfo/providers/{provider}/regions/{region}/prices/{instance_type}"
+
+    def _get_azure_region_key(self, region):
+        return region.replace(" ", "").lower()
```

Comment:
Not sure if this is applicable here, but I remember in the Azure instance price scraper we do some cleaning when it comes: to region names: https://github.com/elotl/kip/blob/master/scripts/create_instance_data/create_azure_instance_data.py#L22-L84